src/rdf4cpp/rdf/Statement.cpp

```diff
@@ -0,0 +1,9 @@
+#include "Statement.hpp"
+namespace rdf4cpp::rdf {
+Statement::Statement(Node subject, Node predicate, Node object) : query::TriplePattern(subject, predicate, object) {}
```

Comment:
same here: maybe validate on creation as well. 

(maybe a factory would be better even though it is less concise)

Reply:
What benefits do we get by using a factory?

Reply:
Using a constructor we cannot validate, because we would need to throw an exception if it wouldn't be valid. 
Using a factory we can validate on the spot and return null or better a Result object, containing the error why it's invalid if it's invalid.

Reply:
(see above